### ISSUE TITLE
Limit the size of `.bsp/mill-bsp.stderr`

### ISSUE THREAD

**Issue by lefou** [com-lihaoyi/mill#2199]
Currently, this file is growing unlimited, which can become an issue if disc space gets low.

**Comment by seroperson** [comment]
I think it should be implemented like some kind of log rotation with specifying somewhere (or just harcoding) values like `logMaxSize`, `logFileLimit`. There are a lot of examples of such problem (the first one which I googled: [link](https://bugs.eclipse.org/bugs/show_bug.cgi?id=87504)) and looks like we should implement something like this.

**Comment by lefou** [comment]
Yeah log-rotation might be nice. This topic seems simple and very common, yet it provides lots of ways to shot yourself in the foot. Especially, as we target different platforms including Windows, which provides lots of stumbling blocks like file locking. Maybe, using some small proven library isn't the worst idea.

**Comment by seroperson** [comment]
[rotating-fos](https://github.com/vy/rotating-fos) looks suitable here.

**Comment by gamlerhart** [comment]
I think this file is not generated anymore? At least I don't see it since Mill 0.12?
